controllers/aws_config_loader.go

```diff
@@ -64,17 +72,17 @@ func (l AwsConfigLoader) LoadAwsConfigWithOverrides(regionOverride string, jobSp
 
 	// If a custom endpoint is requested, install custom resolver for SageMaker into config.
 	if customEndpoint != "" {
-		customSageMakerResolver := func(service, region string) (aws.Endpoint, error) {
-			if service == sagemaker.EndpointsID {
-				return aws.Endpoint{
+		customSageMakerResolver := func(service, region string, optFns ...func(*awsendpoints.Options)) (awsendpoints.ResolvedEndpoint, error) {
+			if service == "sagemaker" {
+				return awsendpoints.ResolvedEndpoint{
```

Comment:
Reminder that we need to run PrivateLink tests

Reply:
btw we do have integ test for custom endpoint which are passing